### PR TITLE
[https://nvbugs/6485885][fix] Stop thinking-budget processor re-forcing the reasoning end tag

### PR DESCRIPTION
**Pull request by Wanli-Jiang**
With the overlap scheduler, logits processors run one step behind the sampled tokens (request.get_tokens() does not yet include the previous iteration's token). ThinkingBudgetLogitsProcessor was stateless and re-derived its decision from that stale view, so the step right after forcing the reasoning end tag could not see it and forced it again. The raw stream then contained a doubled end tag (e.g. '</think></think>'); the reasoning parser consumes the first and the second leaks into message.content (reproduced 3/3 with nano-v3 + thinking_token_budget on nemotron-nano-3.5; clean with the overlap scheduler disabled).

Track per-(request, beam) progress through the end sequence: continue from recorded progress instead of re-deriving from token_ids, stop once the sequence is complete, and reset when the view shows the block closed so later reasoning blocks are budgeted again. Also fixes multi-token end sequences restarting at the first token under the same lag.

### Dev Engineer Review

- Updated `ThinkingBudgetLogitsProcessor` to track per-request, per-beam reasoning-end progress, preventing duplicated end tags when the overlap scheduler exposes stale token views.
- Supports multi-token reasoning-end sequences and resets tracking after the reasoning block closes.
- Preserves stream handling while explicitly ignoring `client_id`.
- No public API or configuration changes.

### QA Engineer Review

- Added regression coverage for stale token views and multi-token reasoning-end progress in `tests/unittest/llmapi/test_sampling_params.py`.
- No test-list changes were identified; coverage is not listed in `tests/integration/test_lists/`.
- **Verdict: needs follow-up** for CI/manual test-list coverage.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/llmapi/thinking_budget.py [hunk 1]
```diff
@@ -33,6 +33,19 @@ def __init__(
...
+        #
+        # We keep this instead of re-deriving from `token_ids` because the overlap
+        # scheduler runs the processor one step behind the sampled tokens: a stale
+        # view would not show the just-forced end sequence, so re-derivation would
+        # force it again and leak a duplicate end tag (e.g. `</think>`).
+        #
+        # Keys are reclaimed only when a block is seen to close, so entries for
+        # blocks that never close persist for the processor's lifetime -- bounded
+        # per request, but a slow drip if one SamplingParams is shared across many.
+        self._end_progress: dict = {}
```
[the remark this answers is not in the document]
Comment (Wanli-Jiang) [reply tensorrt_llm/llmapi/thinking_budget.py:48]: Don't add eviction machinery — it'd add complexity for a narrow, low-severity drip, against the grain of the
  just-completed /simplify pass. Reply to CodeRabbit acknowledging it's known and documented (the comment it
  quoted), noting the interface has no completion hook and the server paths are already request-scoped. If you
  want to fully silence it with a genuinely cheap guard, the lightweight option is a size cap on the dict — but
  I'd only do that if a reviewer insists.

**Review by JunyiXu-nv — APPROVED** [review]
LGTM!
